snuba/query/processors/impact_processor.py

```diff
@@ -0,0 +1,88 @@
+from snuba.query.conditions import (
+    binary_condition,
+    BooleanFunctions,
+    ConditionFunctions,
+)
+from snuba.query.expressions import (
+    Expression,
+    FunctionCall,
+    Literal,
+)
+from snuba.query.dsl import div, multiply, plus, minus, countIf, count
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class ImpactProcessor(QueryProcessor):
+    """
+    Resolves the impact function call into
+
+    impact = (1 - (apdex({col}, {satisfied}))) + ((1 - (1 / sqrt(uniq({user_col})))) * 3)
```

Comment:
Would this be easier if Apdex was part of the DSL?

Reply:
This applies to Apdex as well, but I wonder if the col / user_col should come from the dataset rather than be provided by the client since passing any other value for duration and user doesn't really make sense.

Reply:
I agree that I don't see any other use case for this right now, but I left it open ended in Snuba because we are adding a layer in Sentry to validate these columns. This way Snuba can assume that the columns are correct and if we end up with different use cases we only need to change the sentry validation layer to allow those cases.

Reply:
I also saw an earlier comment from Filippo that mentioned wanting to keep the DSL as limited as possible, so I kept it that way. I really think impact is the only function that will rely on apdex.

Reply:
I would not add it to the general dsl, since it is specific to some datasets. Though it would make sense to have a convenience method that you can use both here and in the Apdex processor so you do not have to copy paste code.
So basically, like adding it to the dsl but not in the dsl.py file.